orc8r/cloud/go/services/bootstrapper/client_api.go

```diff
@@ -0,0 +1,96 @@
+/*
+Copyright 2020 The Magma Authors.
+
+This source code is licensed under the BSD-style license found in the
+LICENSE file in the root directory of this source tree.
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package bootstrapper
+
+import (
+	"context"
+
+	"magma/orc8r/lib/go/protos"
+
+	merrors "magma/orc8r/lib/go/errors"
+	"magma/orc8r/lib/go/registry"
+
+	"github.com/golang/glog"
+	"google.golang.org/grpc/codes"
+	"google.golang.org/grpc/status"
+)
+
+func GetToken(ctx context.Context, networkID string, logicalID string, refresh bool) (string, error) {
+	client, err := getCloudRegistrationClient()
+	if err != nil {
+		return "", err
+	}
+
+	req := &protos.GetTokenRequest{
+		GatewayDeviceInfo: &protos.GatewayDeviceInfo{
+			NetworkId:            networkID,
+			LogicalId:            logicalID,
+		},
+		Refresh:                refresh,
+	}
+
+	res, err := client.GetToken(ctx, req)
+	return res.Token, err
+}
+
+func GetGatewayRegistrationInfo(ctx context.Context, token string) (*protos.GetGatewayRegistrationInfoResponse, error) {
+	client, err := getCloudRegistrationClient()
+	if err != nil {
+		return nil, err
+	}
+
+	req := &protos.GetGatewayRegistrationInfoRequest{}
+
+	res, err := client.GetGatewayRegistrationInfo(ctx, req)
+	if err != nil {
+		return nil, err
+	}
+
+	return res, nil
+}
+
+func GetGatewayDeviceInfo(ctx context.Context, token string) (*protos.GatewayDeviceInfo, error) {
+	client, err := getCloudRegistrationClient()
+	if err != nil {
+		return nil, err
+	}
+
+	req := &protos.GetGatewayDeviceInfoRequest{
+		Token:                token,
+	}
+
+	res, err := client.GetGatewayDeviceInfo(ctx, req)
+	if err != nil {
+		return nil, err
+	}
+
+	clientErr := res.Response.(*protos.GetGatewayDeviceInfoResponse_Error)
+	if clientErr != nil{
+		// TODO(reginawang3495): Be more precise based on the different errors?
+		return nil, status.Error(codes.Unauthenticated, clientErr.Error)
+	}
+
+	gatewayInfo := res.Response.(*protos.GetGatewayDeviceInfoResponse_GatewayDeviceInfo)
+	return gatewayInfo.GatewayDeviceInfo, nil
```

Comment:
You can switch on the type here to make things more self-contained https://github.com/magma/magma/blob/509e5b4bcdb68b0021044fb1dc8cf47554446b8f/orc8r/cloud/go/services/metricsd/exporters/proto_helpers.go#L36-L48

And be sure to always include a default for the switch statement!